bot.js

```diff
@@ -68,20 +73,25 @@ function saySomething() {
   var words = message.split(' ')
   
   // Iterate over all rules and find the first one that matches
-  for(var i = 0; i<rules.length; i++) 
+  for (var i = 0; i<rules.length; i++) 
   {
-  var match = matchPattern(rules[i].pattern, words, 0, 0)
-  if (match) {
-  // Reconstruct a reply - if a token is a number, find the
-  // matched word from 'match', otherwise just append the word
-  var res = ""
-  for(var token of rules[i].reply) {
-    if (typeof(token) == "number") res += match[token]
-      else res += token;
-  }
-  answer = res;
-  }
+  	var match = matchPattern(rules[i].pattern, words, 0, 0);
+  	if (match) {
+  	// Reconstruct a reply - if a token is a number, find the
```

Comment:
I would indent the comment so that it is alogned with the body (i.e. the following `var res = .. ` line).